pkg/controllers/user/nodesyncer/cordonfieldssyncer.go

```diff
@@ -0,0 +1,32 @@
+package nodesyncer
+
+import (
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func (m *NodesSyncer) syncCordonFields(key string, obj *v3.Node) error {
+	if obj == nil {
+		return nil
+	}
+	machine := obj.DeepCopy()
```

Comment:
why are we doing a deep copy? it should be done only when we update the machine, and we should be scarce and do it only when something gets changed on the update